src/stubs.ts

```diff
@@ -1,29 +1,37 @@
-import { transformVNodeArgs, h, createVNode } from 'vue'
+import {
+  transformVNodeArgs,
+  h,
+  ComponentPublicInstance,
+  Slots,
+  ComponentOptions,
+  defineComponent,
+  VNodeProps,
+  VNodeTypes
+} from 'vue'
 import { hyphenate } from './utils/vueShared'
 import { MOUNT_COMPONENT_REF, MOUNT_PARENT_NAME } from './constants'
 import { config } from './config'
 import { matchName } from './utils/matchName'
 
-interface IStubOptions {
+interface StubOptions {
   name?: string
   props: any
 }
 
-type VNodeArgs = Parameters<typeof createVNode>
-
-function getSlots(ctx) {
+function getSlots(ctx: ComponentPublicInstance): Slots | undefined {
   return !config.renderStubDefaultSlot ? undefined : ctx.$slots
 }
 
-const createStub = ({ name, props }: IStubOptions) => {
+const createStub = ({ name, props }: StubOptions): ComponentOptions => {
   const anonName = 'anonymous-stub'
   const tag = name ? `${hyphenate(name)}-stub` : anonName
 
-  const render = (ctx) => {
-    return h(tag, {}, getSlots(ctx))
+  const render = (ctx: ComponentPublicInstance) => {
+    // TS is not happy with this signature, so we cast the Slots as any
+    return h(tag, {}, getSlots(ctx) as any)
```

Comment:
TS is not happy here, as there is no signature of `h` matching `h(string, {}, Slots)`. But it works, so I added `as any` even if I don't like it...

Reply:
After digging a bit into it, I think this should be fixed in `vue-next`, and I submitted a PR: https://github.com/vuejs/vue-next/pull/1293
We can merge as is for now, and I'll remove the `as any` cast if Evan accepts my PR.